dw-surface.js

```diff
@@ -249,12 +251,32 @@ export class DwSurface extends LitElement {
     super.connectedCallback();
     this.addEventListener("mousedown", this._onMouseDown);
     this.addEventListener("blur", this._onBlur);
+    this._debounceResizeObserver = debounce(this._debounceResizeObserver.bind(this), 200);
+    this.resizeObserver = new ResizeObserver(() => this._debounceResizeObserver());
+    this.resizeObserver.observe(this);
   }
 
   disconnectedCallback() {
     super.disconnectedCallback();
     this.removeEventListener("mousedown", this._onMouseDown);
     this.removeEventListener("blur", this._onBlur);
+    this.resizeObserver.unobserve(this);
+  }
+
+  _debounceResizeObserver() {
+    this._computeScrollerHeight()
+  }
+
+  _computeScrollerHeight() {
+    let scrollerEl = this.renderRoot.querySelector(".scroller");
+    let paddingTop = parseInt(window.getComputedStyle(this).getPropertyValue("padding-top"));
+    let paddingBottom = parseInt(window.getComputedStyle(this).getPropertyValue("padding-bottom"));
+
+    let scrollerElheight = scrollerEl.offsetHeight + paddingTop + paddingBottom;
+
+    if (this.offsetHeight !== scrollerElheight) {
```

Comment:
This logic has many performance issues & complications. So, needs to be re-written.